### PR TITLE
Hook up remaining DSA notifications

### PR DESCRIPTION
**Pull request by nick-funk**
## What does this PR do?

- Hooks up the DSA notifications that happen when a user's comment is deemed illegal
- Hooks up DSA reason information when comments are rejected into notifications

## These changes will impact:

- [X] commenters
- [X] moderators
- [ ] admins
- [ ] developers

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices.

## How do I test this PR?

DSA Reporting:
- Create a comment
- Report it with the DSA reporting
- Process the comment with an admin/mod user in the DSA report flow
- Decide the comment is illegal in the DSA report flow
    - Specify the grounds and explanation
- Check the original user's notifications
    - See a new notification showing the illegal reasoning, grounds, explanation, and the referenced comment
- Check the reporting user's notifications
    - See a new notification indicating that the report decision was made

General Approve/Reject of comments with reasons:
- Create a comment
- Reject it using an admin/moderator account in the mod queues
    - Provide a reason
- Check notifications of the original commenter
    - See that they have a notification indicating the rejection code and other legal details

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

No special considerations, it goes into the DSA epic branch when complete.

### CODE REVIEW

File: server/src/core/server/services/dsaReports/reports.ts [hunk 1]
```diff
@@ -201,8 +205,22 @@ export async function makeDSAReportDecision(
       commentID,
       commentRevisionID,
       userID,
-      now
+      now,
+      undefined,
+      false
     );
+
+    if (comment.authorID) {
```
Comment (kabeaty) [inline server/src/core/server/services/dsaReports/reports.ts]: We need to send out a notification to the reporter of the DSA report of the decision as well here, right?

File: client/src/core/client/stream/tabs/Notifications/NotificationContainer.tsx [hunk 2]
```diff
@@ -35,6 +36,9 @@ const NotificationContainer: FunctionComponent<Props> = ({
     return createdAtDate.getTime() <= lastSeenDate.getTime();
   }, [createdAt, viewer]);
 
+  // eslint-disable-next-line no-console
+  console.log(body);
```
Comment (kabeaty) [inline client/src/core/client/stream/tabs/Notifications/NotificationContainer.tsx]: Just noting here that this needs to be removed still.

**Review by kabeaty — COMMENTED** [review]
Looking great!! Just one comment about sending a notification to the reporter too.

File: server/src/core/server/services/dsaReports/reports.ts (hunk 1 above)
Comment (nick-funk) [reply server/src/core/server/services/dsaReports/reports.ts]: This is hooked up now, I just initially posted my work as a draft to keep visibility for this work available to y'all!

File: client/src/core/client/stream/tabs/Notifications/NotificationContainer.tsx (hunk 2 above)
Comment (nick-funk) [reply client/src/core/client/stream/tabs/Notifications/NotificationContainer.tsx]: this has been removed! 👍

**Review by kabeaty — APPROVED** [review]
Looking good!!

**Pull request merged**